src/realm/query_engine.hpp

```diff
@@ -512,6 +512,13 @@ class IntegerNode<LeafType, Equal> : public IntegerNodeBase<LeafType> {
             if (const int64_t* val = it->get_if<int64_t>()) {
                 m_needles.insert(*val);
             }
+            else if (const double* val = it->get_if<double>()) {
+                // JS encodes numbers as double
+                m_needles.insert(int64_t(*val));
```

Comment:
Value should only be inserted if the integer part equals the double value (remainder is zero). I would have expeted this conversion to happen in driver.cpp. Did not have the energy to investigate.

Reply:
Good point. I update the code here to do the same. The conversion you are thinking of happens [here](https://github.com/realm/realm-core/blob/master/src/realm/parser/driver.cpp#L1416-L1446) and only applies when the constants are embedded in the query string. In the case that an argument list is read, then we use Mixed numeric comparison which ends up doing integer equality. That seems to be a subtle but preexisting problem that I don't think I should attempt to fix here.